src/genotype.cpp

```diff
@@ -116,10 +122,14 @@ Genotype::Genotype(const std::filesystem::path& model_file){
                 });
         }
 
-        /**
-         * FIXME: After adding the graph representation of the network,
-         * FIXME: also construct the graph HERE!
-         */
+        // construct graph representation of the initial network
+        for(Connection connection : connection_genes){
```

Comment:
You don't need to loop two times. The following code is all you need:
```
// construct graph representation of the initial network
        for(Connection connection : connection_genes)
                if(connection.enable)
                        graph[connection.in].insert(std::make_pair(connection.out, connection.weight));
```
Simply extract this code to a standalone function as mentioned above.